python/cudf/cudf/core/dataframe.py

```diff
@@ -6329,6 +6329,45 @@ def explode(self, column, ignore_index=False):
 
         return super()._explode(column, ignore_index)
 
+    def pct_change(
+        self, periods=1, fill_method="ffill", limit=None, freq=None
+    ):
+        """
+        Calculates the percent change between sequential elements
+        in the DataFrame.
+
+        Parameters
+        ----------
+        periods : int, default 1
+            Periods to shift for forming percent change.
+        fill_method : str, default 'ffill'
+            How to handle NAs before computing percent changes.
+        limit : int, optional
+            The number of consecutive NAs to fill before stopping.
+            Not yet implemented.
+        freq : str, optional
+            Increment to use from time series API.
+            Not yet implemented.
+
+        Returns
+        -------
+        DataFrame
+        """
+        if limit is not None:
+            raise NotImplementedError("limit parameter not supported yet.")
+        if freq is not None:
+            raise NotImplementedError("freq parameter not supported yet.")
+        elif fill_method not in {"ffill", "pad", "bfill", "backfill"}:
+            raise ValueError(
+                "fill_method must be one of 'ffill', 'pad', "
+                "'bfill', or 'backfill'."
+            )
+
+        data = self.fillna(method=fill_method, limit=limit)
```

Comment:
Does this always create a copy? is it possible to avoid copy by checking something like `has_na`?

Reply:
Could that perhaps happen upstream in the `fillna` function? I could imagine a short circuit based on the null_count

Is there any ecosystem behavior that depends on `fillna` always returning a copy, even if there are no nulls?

Reply:
@karthikeyann @beckernick @shwina According to this thread , looks like there's still pending/ ongoing discussions about optimizing `fillna() `
I was wondering if we can proceed with my current code as is, then optimize in a follow-up PR?